### PR TITLE
Add underfs interfaces

### PR DESCRIPTION
**Pull request by yyongycy**
### What changes are proposed in this pull request?

Add underfs interfaces.

### Why are the changes needed?

Add underfs interfaces for extentions.

### Does this PR introduce any user facing changes?
NA

### CODE REVIEW

File: dora/core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java [hunk 1]
```diff
@@ -1359,4 +1359,108 @@ private RetryPolicy getRetryPolicy() {
   private RetryPolicy getRetryOncePolicy() {
     return new CountingRetry(1);
   }
+
+  @Override
+  public String initObjectMultiPart(String path,
```
Comment (yyongycy) [inline dora/core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java:1364]: These files are from @Jackson-Wang-7

File: dora/core/common/src/main/java/alluxio/underfs/options/ListMultiPartOptions.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Method options for listMultipartUploads in {@link alluxio.underfs.UnderFileSystem}.
+ */
+
+@PublicApi
+@NotThreadSafe
+public class ListMultiPartOptions {
```
Comment (yyongycy) [inline dora/core/common/src/main/java/alluxio/underfs/options/ListMultiPartOptions.java:24]: From @Jackson-Wang-7

**Review by Jackson-Wang-7 — APPROVED** [review]
lgtm

**Comment by yyongycy** [comment]
alluxio-bot, merge this please

**Pull request merged**